python/example/berry/test_berry_kevlar.py

```diff
@@ -0,0 +1,94 @@
+import fast_inla
+import numpy as np
+import pykevlar
+from binomial_accumulate import binomial_accumulator
+from pykevlar.driver import accumulate_process
+from pykevlar.grid import HyperPlane, make_cartesian_grid_range
+from pykevlar.model.binomial import BerryINLA2
+from scipy.special import logit
+
+
+def test_kevlar_and_py_binomial_accumulate():
+    n_arms = 2
+    n_arm_samples = 35
+    seed = 10
+    n_theta_1d = 16
+    sim_size = 100
+    # getting an exact match is only possible with n_threads = 1 because
+    # parallelism in the kevlar accumulator leads to a different order of random
+    # numbers.
+    n_threads = 1
+
+    # define null hypos
+    null_hypos = []
+    for i in range(n_arms):
+        n = np.zeros(n_arms)
+        # null is:
+        # theta_i <= logit(0.1)
+        # the normal should point towards the negative direction. but that also
+        # means we need to negate the logit(0.1) offset
+        n[i] = -1
+        null_hypos.append(HyperPlane(n, -logit(0.1)))
+    gr = make_cartesian_grid_range(
+        n_theta_1d, np.full(n_arms, -3.5), np.full(n_arms, 1.0), sim_size
+    )
+    gr.create_tiles(null_hypos)
+    gr.prune()
+    n_tiles = gr.n_tiles()
+
+    fi = fast_inla.FastINLA(2)
+
+    # Run the C++ INLA Berry inference/rejection via accumulate_process.
+    b = BerryINLA2(
+        n_arm_samples,
+        [0.85],
+        np.full(2, fi.thresh_theta),
+        fi.sigma2_rule.wts.copy(),
+        fi.cov.reshape((-1, 4)).T.copy(),
+        fi.neg_precQ.reshape((-1, 4)).T.copy(),
+        fi.logprecQdet.copy(),
+        fi.log_prior.copy(),
+        fi.tol,
+        fi.logit_p1,
+    )
+    out = accumulate_process(b, gr, sim_size, seed, n_threads)
+
+    # Use the mt19937 object exported from C++ so that we can match the C++ random
+    # sequence exactly. This is not necessary in the long term but is temporarily
+    # useful to ensure that this code is producing identical output to the C++
+    # version.
+    n_arm_samples = 35
+    gen = pykevlar.mt19937(seed)
+
+    # We flip the order of n_arms and n_arm_samples here so the random number
+    # generator produces the same sequence of uniforms as are used in the C++ kevlar
+    # internals. The Kevlar function operates in column-major/Fortran order. Whereas
+    # here, numpy operates in row-major/C ordering b
+    samples = np.empty((sim_size, n_arms, n_arm_samples))
+    gen.uniform_sample(samples.ravel())
+    # after transposing, samples will have shape (sim_size, n_arm_samples, n_arms)
+    samples = np.transpose(samples, (0, 2, 1))
+
+    theta = gr.thetas().T.copy()
+    # TODO: it'd be nice to add theta_tiles and is_null_per_arm to the GridRange object!
```

Comment:
I disagree - C++ side should be minimal in memory footprint as possible. This is only needed in Python because it's pythonic to make a full array of things before processing. I would only agree if we start needing this if enough C++ models require it. So far, including direct bayes, no C++ model ever requires this extra allocation.

Reply:
I also look at this as the former to be clear. But I still want the c++ side to be a very minimal self-contained set of tools as well. I don't think we should put things in there solely because it's convenient for python unless you have no choice. So in the case where we wanted to add things specifically for python, we can just wrap the c++ class. It's not the performance, it's memory.

Reply:
What about an accessor method on GridRange that constructs the array on the C++ side on demand but doesn't store it?

Reply:
Regardless, this is just a todo comment here and we can ignore it or I can delete it. Nothing needs to happen right now.